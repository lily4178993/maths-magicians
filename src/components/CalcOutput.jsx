import React from 'react';
import PropTypes from 'prop-types';

const CalcOutput = ({ outputValue }) => (
  <div className="output">
    <div>{outputValue}</div>
  </div>
);

CalcOutput.propTypes = {
  outputValue: PropTypes.string.isRequired,
};

export default CalcOutput;
