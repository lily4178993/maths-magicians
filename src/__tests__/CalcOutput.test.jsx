import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import CalcOutput from '../components/CalcOutput';

test('PropTypes snapshot', () => {
  const { asFragment } = render(<CalcOutput outputValue="123" className="output" />);
  expect(asFragment()).toMatchSnapshot();
});

test('Simulate user interaction', () => {
  const { getByText } = render(<CalcOutput outputValue="123" className="output" />);
  const outputElement = getByText('123');
  expect(outputElement).toBeInTheDocument();
});
