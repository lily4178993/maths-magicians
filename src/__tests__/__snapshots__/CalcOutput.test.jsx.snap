// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropTypes snapshot 1`] = `
<DocumentFragment>
  <div
    class="output"
  >
    123
  </div>
</DocumentFragment>
`;
